cmd/manager/scap.go

```diff
@@ -520,7 +586,47 @@ func fetch(ctx context.Context, streamDispatcher streamerDispatcherFn, rfClients
 			return nil, warnings, err
 		}
 	}
-	return results, warnings, nil
+	results, warnings, err := saveConsistentKubeletResult(results, warnings)
+	return results, warnings, err
+}
+
+// Only save consistent KubeletConfigs per node role.
+func saveConsistentKubeletResult(result map[string][]byte, warning []string) (map[string][]byte, []string, error) {
+	if len(result) == 0 {
+		return result, warning, nil
+	}
+	kubeletConfigsRole := make(map[string][]byte)
+	for DumpPath, content := range result {
+		role, node := getRoleNodeNameFromDumpPath(DumpPath)
+		if role == "" {
+			continue
+		}
+		if existingKC, ok := kubeletConfigsRole[role]; ok {
+			diff, err := jsondiff.CompareJSON(existingKC, content)
```

Comment:
I like the idea of only saving intersection! This way if something is different and we care, then we find out because the attribute will not be there. Nice!